web-common/src/features/dashboards/stores/filter-utils.ts

```diff
@@ -253,3 +254,47 @@ export const sanitiseExpression = (
   if (!where?.cond?.exprs?.length) return undefined;
   return where;
 };
+
+export function isExpressionIncomplete(expression: V1Expression): boolean {
+  // Check if the operation is unspecified at any level of the condition.
+  function isOperationUnspecified(cond: V1Condition): boolean {
```

Comment:
Any reason fro this to be inside the `isExpressionIncomplete` method?

Reply:
Same with `isValDefinedAndNonEmpty`

Reply:
No strong preference – just moved them outside